### ISSUE TITLE
Do not store the entire CAR raw blocks in memory

### ISSUE THREAD

**Issue by masih** [data-preservation-programs/singularity#147]
The current dataset worker seems to store the entire raw block data from a CAR file in memory? 

https://github.com/data-preservation-programs/singularity/blob/0b8005d3068dfd78bb4eea84316ee12b2d58f7a8/service/datasetworker/daggen.go#L76-L81

**Comment by xinaxu** [comment]
That's dumping the folder structure into a CAR file. This is a problem for dataset that contains >200 million files. 
For motion it is not going to be used because it is motion does not use "daggen".

I'll make this a low-priority item for now.

**Comment by masih** [comment]
> motion does not use "daggen"

@hannahhoward is that consistent with your understanding?

**Comment by anjor** [comment]
@masih yeah, daggen is used when you have a directory structure. Given that the motion api only deals with blobs it should not be using daggen.
